Text/CustomEntitySearch/CustomEntityLookupImplementation.cs

```diff
@@ -0,0 +1,289 @@
+                    {
+                        var normalizedSubstring = NormalizeWord(substring, selection.CaseSensitive, selection.AccentSensitive);
+                        var normalizedWord = selection.GetNormalizedText(NormalizeWord);
+                        var editDistance = CustomEntityLookupEditDistanceHelper.CalculateDamerauLevenshteinDistance(normalizedSubstring, normalizedWord);
+                        var fuzziness = selection.FuzzyEditDistance >= normalizedWord.Length ? normalizedWord.Length - 1 : selection.FuzzyEditDistance;
+                        fuzziness = Math.Max(0, fuzziness);
+                        if (editDistance <= fuzziness)
+                        {
+                            if (!entitiesFound.ContainsKey(selection.ParentEntityReference))
+                            {
+                                entitiesFound[selection.ParentEntityReference] = new FoundEntity(
+                                    name: selection.ParentEntityReference.Name,
+                                    description: selection.ParentEntityReference.Description,
+                                    id: selection.ParentEntityReference.Id,
+                                    type: selection.ParentEntityReference.Type,
+                                    subtype: selection.ParentEntityReference.Subtype,
+                                    matches: new List<FoundEntityMatchDetails>());
+                            }
+
+                            entitiesFound[selection.ParentEntityReference].Matches.Add(
+                                new FoundEntityMatchDetails(
+                                    text: substring,
+                                    offset: startIndex,
+                                    length: substring.Length,
+                                    matchDistance: editDistance));
+                        }
+                    }
+                }
+            }
+
+            return entitiesFound.Values.ToList();
+        }
+
+        private string BuildRegexForExactMatch(CustomEntitySelection selection)
+        {
+            string word = selection.Text;
+            StringBuilder escapedWord = new StringBuilder(@"(?=(");
+            if (!word.First().IsDelineating())
+            {
+                escapedWord.Append(@"\b");
+            }
+
+            for (int currWordCharIndex = 0; currWordCharIndex < word.Length; currWordCharIndex++)
+            {
+                if (word[currWordCharIndex].IsDelineating())
+                {
+                    escapedWord.Append(@"\b");
+                }
+                else
+                {
+                    escapedWord.Append(word[currWordCharIndex]);
+                }
+            }
+            if (!word.Last().IsDelineating())
+                escapedWord.Append(@"\b");
+            escapedWord.Append("))");
+            string pattern = (selection.CaseSensitive) ? @"(?x)" + escapedWord : @"(?ix)" + escapedWord;
```

Comment:
Just return it.

Reply:
compiler will optimize this variable away. Keeping it before returning makes debugging easier since i can see the intermediate state in debug build

---
In reply to: [337797973](https://github.com/Azure-Samples/azure-search-power-skills/pull/15#discussion_r337797973) [](ancestors = 337797973)